python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -1089,6 +1105,20 @@ def sample_to_pdf(self,
         return pdf
 
 
+class LazySparkXShards(SparkXShards):
+    def __init__(self,
+                 rdd: Union["PipelinedRDD", "RDD"],
+                 class_name: str = None) -> None:
+        super(LazySparkXShards, self).__init__(rdd, True, class_name)
+        self.user_cached = True
```

Comment:
Change it to `user_caching`?

And you need to set `self.eager` to false

Reply:
Seems in our logic if transient is True, then eager is always False.